lib/src/main/java/org/team1540/rooster/drive/pipeline/DriveData.java

```diff
@@ -0,0 +1,85 @@
+package org.team1540.rooster.drive.pipeline;
+
+import java.util.Objects;
+import java.util.OptionalDouble;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Encapsulates data for one side of a tank drivetrain.
+ *
+ * {@code DriveData} instances are typically passed around as members of {@link TankDriveData}
+ * instances in drive pipelines. Each {@code DriveData} instance contains values (or empty {@link
+ * OptionalDouble OptionalDoubles}) for  {@linkplain #position}, {@linkplain #velocity}, {@linkplain
+ * #acceleration}, and {@linkplain #additionalFeedForward feed-forward}.
+ *
+ * @see TankDriveData
+ */
+@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+public class DriveData {
+
+  /**
+   * The desired position in position-units, or an empty optional if velocity should not be
+   * controlled.
+   */
+  @NotNull
+  public final OptionalDouble position;
+  /**
+   * The desired velocity in position-units per second, or an empty optional if velocity should not
+   * be controlled.
+   */
+  @NotNull
+  public final OptionalDouble velocity;
```

Comment:
Do we want our position derivates to be hard coded, or do we want a more flexible system?

Reply:
eh ok